sonic-net/sonic-buildimage
# Update macsec CAK keys in profile for tests to change to type7 encoded format

#### Why I did it
The changes needed in various macsec unit tests and config plugin when we move to accept the type 7 encoded key format for macsec. This goes along with PR : https://github.com/sonic-net/sonic-swss/pull/2892 raised earlier.

##### Work item tracking
- Microsoft ADO **(number only)**: 25046448

#### How I did it
Updated the macsec profiles used in unit test with the type7 encoded CAK keys.

Updated the length check in macsec config plugin to accept not strings of length 66 and 130 bytes depending on whether it is 128 byte cipher suite or 256 byte cipher suite

Additional changes needed to the yang model file. 
1. Increase the CAK key length from 32 bytes and 64 bytes with plain text format --> 66 bytes and 130 bytes with type 7 encoded keys for 128 byte and 256 byte cipher suites.
2. Removing the check to validate the length of CAK and CKN, the key and name need not be same length

#### How to verify it
Validated with CLI commands to make sure macsec profiles are created correctly.
More tests added in PRs : https://github.com/sonic-net/sonic-swss/pull/2892, https://github.com/sonic-net/sonic-mgmt/pull/9812

#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
